src/quo/components/wallet/swap_input/view.cljs

```diff
@@ -0,0 +1,118 @@
+      [:network-tag-props {:optional true} [:maybe :map]]
+      [:on-change-text {:optional true} [:maybe fn?]]
+      [:enable-swap? {:optional true} [:maybe :boolean]]
+      [:on-swap-press {:optional true} [:maybe fn?]]
+      [:on-token-press {:optional true} [:maybe fn?]]
+      [:on-max-press {:optional true} [:maybe fn?]]
+      [:customization-color {:optional true} [:maybe :schema.common/customization-color]]
+      [:container-style {:optional true} [:maybe :map]]]]]
+   :any])
+
+(defn view-internal
+  [{:keys [type status token value fait-value show-approval-label? error? network-tag-props
+           approval-label-props default-value enable-swap?
+           currency-symbol on-change-text
+           container-style on-swap-press on-token-press on-max-press]}]
+  (let [theme             (quo.theme/use-theme)
+        pay?              (= type :pay)
+        disabled?         (= status :disabled)
+        loading?          (= status :loading)
+        controlled-input? (some? value)
+        input-ref         (rn/use-ref-atom nil)
+        set-input-ref     (rn/use-callback (fn [ref] (reset! input-ref ref)) [])
+        focus-input       (rn/use-callback (fn [] (.focus ^js @input-ref))
+                                           [input-ref])]
+    [rn/view
+     {:style               container-style
+      :accessibility-label :swap-input}
+     [rn/view {:style (style/content theme)}
+      [rn/view
+       {:style (style/row-1 loading?)}
+       [rn/pressable {:on-press on-token-press}
+        [token-avatar/view
+         {:type  :asset
+          :token token}]]
+       (when loading? [rn/view {:style (style/row-1-loader theme)}])
+       (when-not loading?
+         [:<>
+          [rn/pressable
+           {:style    style/input-container
+            :on-press focus-input}
+           [rn/text-input
+            (cond-> {:ref                    set-input-ref
+                     :style                  (style/input disabled? error? theme)
+                     :placeholder-text-color (colors/theme-colors colors/neutral-40
+                                                                  colors/neutral-50
+                                                                  theme)
+                     :input-mode             :decimal
+                     :keyboard-type          :decimal-pad
+                     :auto-focus             true
+                     :on-change-text         on-change-text
+                     :editable               (not disabled?)
+                     :default-value          default-value
+                     :placeholder            "0"}
+              controlled-input? (assoc :value value))]
+           [text/text
+            {:size   :paragraph-2
+             :weight :semi-bold
+             :style  {:margin-bottom 1
+                      :color         (colors/theme-colors colors/neutral-50 colors/neutral-40 theme)}}
+            token]]
```

Comment:
Worried about this implementation because of the issues we faced with the `token-input`

Could you please add a video to the PR description where you type in this input.

Usually that `token` text at the end makes the input to flicker :eyes:

Reply:
seems this is a new component @ulisesmac - better if the designs are just made to provide an easier path to bug free implementations.

Reply:
Hey @ulisesmac,
On iOS, it seems to be working fairly well, but on Android, it looks weird. The token input component has the same issues. Has anyone investigated this issue yet? If not, I can look into it and see if we can find a solution. Alternatively, should we ask the designers to come up with a different approach?

iOS:

https://github.com/user-attachments/assets/fd0e005b-43b7-4326-81b4-1e5aef86e7a8

Android:

[Android.webm](https://github.com/user-attachments/assets/7e4e5fd9-4f58-463f-bdfe-5591d87ac439)

Reply:
At this point, this component will only be used in the swap feature, which is entirely under the swap feature flag. Therefore, I think we can address this issue in a separate follow-up PR. Additionally, we need to fix this for the token input component as well. Do you know if we already have an issue for this? If not, I can create one.

Reply:
follow up issue: https://github.com/status-im/status-mobile/issues/20814